apple_passkit/pubspec.yaml

```
@@ -8,17 +8,17 @@ topics:
   - pkpass
 
 environment:
-  sdk: '>=3.4.3 <4.0.0'
-  flutter: 3.22.2
+  sdk: ">=3.4.3 <4.0.0"
+  flutter: ">=3.22.2"
```

```suggestion
  flutter: 3.22.2
```
I prefer working with fixed flutter versions since it simplifies migrations and version bumps

---

While I agree, this doesn't work out for libraries. Consider this example:
Imagine two dependencies which specify for dependency A `flutter: 3.22.2` and for dependency B `flutter: 3.22.1`. Now a consumer tries to use dependency A and B and fails, since the dependencies can't be resolved since they both require the consumer to use two different Flutter versions.

For libraries, version ranges are king, since they make it possible to resolve dependencies for a wide range of varying dependency constraints.

Also, the `dart pub publish` command actively warns you against pinning the Dart/Flutter version to a specific non-range version. I also imagine you're loosing pub points, although I have to admit I have never tried that.

That's also the reason why these libraries specified a rather old Flutter version to start with, since it allows consumers on older Flutter/Dart versions.

---

that makes sense, I'm fine with using these ranges then!